pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -429,6 +454,10 @@ public void close() throws PulsarServerException {
             state = State.Closed;
             isClosedCondition.signalAll();
 
+            CompletableFuture<Void> shutdownFuture =
+                    CompletableFuture.allOf(asyncCloseFutures.toArray(new CompletableFuture[0]));
+            closeFutureReference.set(shutdownFuture);
+            return shutdownFuture;
         } catch (Exception e) {
             if (e instanceof CompletionException && e.getCause() instanceof MetadataStoreException) {
                 throw new PulsarServerException(MetadataStoreException.unwrap((CompletionException) e));
```

Comment:
what about doing this way ?
`return FutureUtil.failedFuture( new PulsarServerException(MetadataStoreException.unwrap((CompletionException) e)));`

the caller expects to receive a failed Future

Reply:
I'm sorry, but I don't really get the point.

Reply:
I've made changes in that direction. please check the most recent changes whether I understood correctly?